pkg/rotator/rotator_test.go

```diff
@@ -179,7 +179,8 @@ func TestReconcile(t *testing.T) {
 	const secretName = "test-secret"
 	const whName = "test-webhook"
 
-	ctx := context.Background()
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
```

Comment:
You'll need to call `cancelFunc` right before `wg.Wait()` - it is waiting for the manager to complete and the manager is waiting for the context to close.
This will not interfere with the `defer` (which still handles early exit) - as It is safe to call `cancelFunc` multiple times.

This is true for the other test as well.

Reply:
@stijndehaes thanks for working on this!  Do you have a sense on when you'd be able to address the above? We have another change that depends on this one and we'd love to see them all get merged :)

Reply:
Sorry missed the previous communication :) Fixed it now

Reply:
Thank you!